xbmc/addons/AddonManager.h

```diff
@@ -267,17 +283,24 @@ namespace ADDON
 
     /*!
      * @brief Get a list of disabled add-on's with info's for the on system
-     * available ones.
+     * available ones with a specific disabled reason.
      *
      * @param[out] addonInfos list where finded addon information becomes stored
      * @param[in] type        The requested type, with "ADDON_UNKNOWN"
      *                        are all add-on types given back who match the case
      *                        with value before.
      *                        If a type id becomes added are only add-ons
      *                        returned who match them. Default is for all types.
+     * @param[in] disabledReason To get all disabled addons use the value
+     *                           "ADDON_DISABLED_REASON_NONE". If any other value
+     *                           is supplied only addons with that reason will be
+     *                           returned.
      * @return true if the list contains entries
      */
-    bool GetDisabledAddonInfos(std::vector<AddonInfoPtr>& addonInfos, TYPE type);
+    bool GetDisabledAddonInfos(
+        std::vector<AddonInfoPtr>& addonInfos,
+        TYPE type,
+        const AddonDisabledReason& disabledReason = ADDON_DISABLED_REASON_NONE);
```

Comment:
I hate default parameters. In most cases they are just for the lazy who extends an exiting method and is not motivated to change the exiting calling code. Do we really need this as a default parameter?

Reply:
It didn’t have the default parameter originally. Alwin why it didn’t as I had two methods instead of one 😐

Reply:
> I hate default parameters (default parameters are pest)

i read this multiple times from multiple guys on this project.. just taking notes ;)